src/core/funcbloc.cpp

```diff
@@ -375,27 +310,38 @@ CIEC_ANY *CFunctionBlock::getVar(CStringDictionary::TStringId *paNameList,
   return nullptr;
 }
 
-CAdapter *CFunctionBlock::getAdapter(CStringDictionary::TStringId paAdapterNameId) const{
+CAdapter *CFunctionBlock::getAdapter(CStringDictionary::TStringId paAdapterNameId) {
   TPortId adpPortId = getAdapterPortId(paAdapterNameId);
 
   if(cgInvalidPortId != adpPortId){
-    return mAdapters[adpPortId];
+    return getAdapterUnchecked(adpPortId);
   }
   return nullptr;
 }
 
-TPortId CFunctionBlock::getAdapterPortId(CStringDictionary::TStringId paAdapterNameId) const{
+const CAdapter *CFunctionBlock::getAdapter(CStringDictionary::TStringId paAdapterNameId) const {
+  TPortId adpPortId = getAdapterPortId(paAdapterNameId);
+
+  if(cgInvalidPortId != adpPortId){
+    return const_cast<CFunctionBlock*>(this)->getAdapterUnchecked(adpPortId);
+  }
+  return nullptr;
+}
+
+TPortId CFunctionBlock::getAdapterPortId(CStringDictionary::TStringId paAdapterNameId) const {
   for(TPortId i = 0; i < mInterfaceSpec->mNumAdapters; ++i){
-    if(mAdapters[i]->getInstanceNameId() == paAdapterNameId){
+    if(const_cast<CFunctionBlock*>(this)->getAdapterUnchecked(i)->getInstanceNameId() == paAdapterNameId){
```

Comment:
with the new adapter api it makes more sense to use here the adapter entry from the interface spec

Reply:
That is much better, thanks.